packages/schema/src/utils/pkg-utils.ts

```diff
@@ -1,20 +1,42 @@
-import fs from 'fs';
-import path from 'path';
+import fs from 'node:fs';
+import path from 'node:path';
 import { execSync } from './exec-utils';
 
 export type PackageManagers = 'npm' | 'yarn' | 'pnpm';
 
-function findUp(names: string[], cwd: string): string | undefined {
-    let dir = cwd;
-    // eslint-disable-next-line no-constant-condition
-    while (true) {
-        const target = names.find((name) => fs.existsSync(path.join(dir, name)));
-        if (target) return target;
-
-        const up = path.resolve(dir, '..');
-        if (up === dir) return undefined; // it'll fail anyway
-        dir = up;
-    }
+/**
+ * A type named FindUp that takes a type parameter e which extends boolean. 
+ * If e extends true, it returns a union type of string[] or undefined. 
+ * If e does not extend true, it returns a union type of string or undefined.
+ * @author Jonathan Stevens (TGTGamer)
```

Comment:
Shall we remove `@author` tag for consistency? I think people usually have a git extension in the IDE that shows the author.

Reply:
Can do, I didn't see any other JSDoc comments so I just left it with my default configuration :)

Reply:
```suggestion
```